lib/bindings/MQTTBinding.js

```diff
@@ -45,85 +45,104 @@ function generateTopics(callback) {
     const topics = [];
 
     config.getLogger().debug(context, 'Generating topics');
-
-    // With leading slashes
-    topics.push(constants.MQTT_SHARE_SUBSCRIPTION_GROUP + '/+/+/' + constants.MEASURES_SUFIX + '/+');
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            '/' +
-            constants.MQTT_TOPIC_PROTOCOL +
-            '/+/+/' +
-            constants.MEASURES_SUFIX +
-            '/+'
-    );
-    topics.push(constants.MQTT_SHARE_SUBSCRIPTION_GROUP + '/+/+/' + constants.MEASURES_SUFIX);
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            '/' +
-            constants.MQTT_TOPIC_PROTOCOL +
-            '/+/+/' +
-            constants.MEASURES_SUFIX
-    );
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            '/+/+/' +
-            constants.CONFIGURATION_SUFIX +
-            '/' +
-            constants.CONFIGURATION_COMMAND_SUFIX
-    );
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            '/' +
-            constants.MQTT_TOPIC_PROTOCOL +
-            '/+/+/' +
-            constants.CONFIGURATION_SUFIX +
-            '/' +
-            constants.CONFIGURATION_COMMAND_SUFIX
-    );
-    topics.push(constants.MQTT_SHARE_SUBSCRIPTION_GROUP + '/+/+/' + constants.CONFIGURATION_COMMAND_UPDATE);
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            '/' +
-            constants.MQTT_TOPIC_PROTOCOL +
-            '/+/+/' +
-            constants.CONFIGURATION_COMMAND_UPDATE
-    );
-
-    //Without leading slashes
-    topics.push(constants.MQTT_SHARE_SUBSCRIPTION_GROUP + '+/+/' + constants.MEASURES_SUFIX + '/+');
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            constants.MQTT_TOPIC_PROTOCOL +
-            '/+/+/' +
-            constants.MEASURES_SUFIX +
-            '/+'
-    );
-    topics.push(constants.MQTT_SHARE_SUBSCRIPTION_GROUP + '+/+/' + constants.MEASURES_SUFIX);
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP + constants.MQTT_TOPIC_PROTOCOL + '/+/+/' + constants.MEASURES_SUFIX
-    );
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            '+/+/' +
-            constants.CONFIGURATION_SUFIX +
-            '/' +
-            constants.CONFIGURATION_COMMAND_SUFIX
-    );
-    topics.push(
-        constants.MQTT_SHARE_SUBSCRIPTION_GROUP +
-            constants.MQTT_TOPIC_PROTOCOL +
-            '/+/+/' +
-            constants.CONFIGURATION_SUFIX +
+    if (config.getConfig().mqtt.sharedSubscriptionsDisabled === true) {
```

Comment:
Why not  
if (config.getConfig().mqtt.sharedSubscriptionsDisabled === true)
   MQTT_SHARE_SUBSCRIPTON_GROUP_VAR = "";
else
  MQTT_SHARE_SUBSCRIPTON_GROUP_VAR = constants.MQTT_SHARE_SUBSCRIPTION_GROUP

and then use the same block but instead of constants.MQTT_SHARE_SUBSCRIPTION_GROUP use MQTT_SHARE_SUBSCRIPTON_GROUP_VAR ?

Reply:
My original thinking was that I wanted to keep as much original code as possible, but the way you described is indeed better and more elegant.  Changed it an tests are passing.